src/pages/billing/Project.js

```diff
@@ -271,7 +306,24 @@ const ProjectDetail = ({ authorizeAndLoadAccounts, billingAccounts, billingProje
         )
       ])
     ]),
-    [spendReportKey]: h(Fragment, [])
+    [spendReportKey]: div({ style: { display: 'grid', gridTemplateColumns: '250px 250px', rowGap: '20px' } }, [
+      div({ style: { gridRowStart: 1, gridColumnStart: 1 } }, [h(IdContainer, [id => h(Fragment, [
+        h(FormLabel, { htmlFor: id }, ['Date range']),
+        h(Select, {
+          id,
+          value: spendReportLengthInDays,
+          options: _.map(days => ({
+            label: `Last ${days} days`,
+            value: days
+          }), [90]),
+          onChange: ({ value: selectedDays }) => {
+            setSpendReportLengthInDays(selectedDays)
+            setTotalCost(null) // This will force the report to be recalculated based on selectedDays
+          }
+        })
+      ])])]),
+      CostCard({ title: 'Total spend', amount: (!!totalCost ? totalCost : '$__.__') })
```

Comment:
We may want to show a more explicit "Retrieving data" message in the future.

Reply:
Alternatively:
```javascript
'$ ...'
```

Reply:
That's a possibility. We have a ticket in our backlog to figure out the best thing to show while loading the data.